alcs-api/src/common/entities/updated-by.subscriber.ts

```diff
@@ -0,0 +1,57 @@
+import { ClsService } from 'nestjs-cls';
+import {
+  DataSource,
+  EntitySubscriberInterface,
+  EventSubscriber,
+  UpdateEvent,
+} from 'typeorm';
+import { UserService } from '../../user/user.service';
+
+export const SYSTEM_ID = 'alcs-api';
+
+@EventSubscriber()
+export class UpdatedBySubscriber implements EntitySubscriberInterface {
+  constructor(
+    dataSource: DataSource,
+    private readonly cls: ClsService,
+    private userService: UserService,
+  ) {
+    dataSource.subscribers.push(this);
+  }
+
+  async beforeInsert(event: UpdateEvent<any>) {
+    if ('auditCreatedBy' in event.entity) {
```

Comment:
I wish we could do something like 'Base.auditCreatedBy.name'

Reply:
Since we can't assume that every Entity extends base, this is the 'correct' way.

Reply:
I'm more from perspective of 'auditCreatedBy' not to be a constant, but that is just typescript limitation